portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,44 @@ function addRandomFunFact() {
   const funFactContainer = document.getElementById('fun-fact-container');
   funFactContainer.innerText = funFact;
 }
+
+/** Slideshow template from https://www.w3schools.com/howto/howto_js_slideshow.asp */
+
+// Initialize slideshow
+var slideIndex = 1;
+showSlides(slideIndex);
+
+/** Slideshow - add next and previous controls */
+function plusSlides(nextOrPrev) {
+  showSlides(slideIndex += nextOrPrev);
+}
+
+/** Slideshow - display image */
+function currentSlide(index) {
+  showSlides(slideIndex = index);
+}
+
+/** Slideshow - display content */
+function showSlides(index) {
+  var i;
+  var slides = document.getElementsByClassName("slides");
+  var dots = document.getElementsByClassName("dot");
+
+  // Handle start and end of slideshow cases
+  if (index > slides.length) {
+    slideIndex = 1;
+  }
+  if (index < 1) {
+    slideIndex = slides.length;
+  }
+
+  // Hide all slides except the current one
+  for (i = 0; i < slides.length; i++) {
+    slides[i].style.display = "none";
+  }
+  for (i = 0; i < dots.length; i++) {
+    dots[i].className = dots[i].className.replace(" dot-active", "");
+  }
```

Comment:
I know most of this is from the W3 tutorial, but it's better not to reuse the same variable i for both loops. It might be better to declare a more descriptive variable inside each of the loops ex: for (var slideListIndex = 0...), for (var dotsIndex = 0...) OR declare the separate variables at the top like you've been doing ex: var slideListIndex; var dotsIndex. Whatever is your personal preference. But using separate variables is generally more readable than reusing the same one (unless they're somehow linked and you're reusing them on purpose, but that's not the case here).

Reply:
Noted! Thanks for pointing this out :)